mainwindow.cc

```diff
@@ -3241,11 +3241,11 @@ void MainWindow::setAutostart(bool autostart)
     }
     reg.sync();
 #else
-    // this is for KDE
+    // this is for Linux
     QString app_fname = QFileInfo(QCoreApplication::applicationFilePath()).baseName();
-    QString lnk(QDir::homePath()+"/.kde/Autostart/"+app_fname);
+    QString lnk(QDir::homePath()+"/.config/autostart/"+"goldendict.desktop");
     if (autostart) {
-        QFile f(QCoreApplication::applicationFilePath());
+        QFile f("/usr/share/applications/goldendict.desktop");
         f.link(lnk);
```

Comment:
My *~/.config/autostart* directory contains copies of desktop files, not links to them. It makes sense because system configuration tools modify these files by adding/removing the `Hidden=true` line to disable/enable app autostart for an individual desktop user.

Since such system configuration tools are readily available in most desktop environments, the autostart feature is not very important on GNU/Linux. An alternative easier to implement fix is to disable or hide this checkbox on Linux.

Reply:
I use myself last KDE session startup option. But some people may not use such DE function, and there is autostart checkbox in Goldendict. So, I think, it will be more logic to leave this function in Goldendict, but make it more correct.
Can You make Goldendict autostart option universal and correct for GNU/Linux?

Thanks in advance.

Reply:
I recommend *copying* the goldendict.desktop file instead of making a link to it.